### ISSUE TITLE
How to evaluate while training

### ISSUE THREAD

**Issue by wazzy** [ftlabs/text-summarization-experiment#1]
``Evaluation should be performed as we train the model, to make sure our model is generalising correctly and is not overfitting on the training set. Running evaluation on unseen data is crucial, to achieve the above and identify when the training process should be stopped.``

How do we evaluate textsum model in between training. Can you please give me some idea regarding that. Do we have to do both the task of training and eval on same machine. 

I ran the evaluation on different machine. I tried doing this after training for 5 days but average loss was not decreasing though training average loss was going down. In 5 days, training reached 500k steps on GPU with 1M articles, 200k validation articles. After 5 days I started evaluation on different machine, it reached near to 100k steps _(tensorboard summary is at same interval, 500k step, so don't know the exact step count) but loss was not at all going down, it was in range of 9 - 10, i.e. from where it started. I considered 200k words for vocab but not lowercase. 

On testing the decode and reference file is also not correct. No relation in reference file and decoded files.
